src/webgl/p5.RendererGL.js

```diff
@@ -32,85 +34,114 @@ defineStrokeJoinEnum('ROUND', 0);
 defineStrokeJoinEnum('MITER', 1);
 defineStrokeJoinEnum('BEVEL', 2);
 
-const lightingShader = readFileSync(
+const getCachedShader= (shaderKey, shaderSource)=>{
+  if(!shaderCache[shaderKey]){
+    shaderCache[shaderKey]=p5.createShader(shaderSource);
+  }
+  return shaderCache[shaderKey];
+};
+
+const lightingShader = getCachedShader('lightingShader',readFileSync(
   join(__dirname, '/shaders/lighting.glsl'),
   'utf-8'
-);
-const webgl2CompatibilityShader = readFileSync(
+));
+const webgl2CompatibilityShader = getCachedShader('webgl2CompatibilityShader',readFileSync(
```

Comment:
I don't think we want to wrap all of these in `getCachedShader` -- these are not full shaders, just pieces of shader source that will turn into a full shader.